test/spec/Feature/Query/CustomMediaSpec.hs

```diff
@@ -382,3 +382,26 @@ spec = describe "custom media types" $ do
       `shouldRespondWith`
       [json| {"code":"PGRST107","details":null,"hint":null,"message":"None of these media types are available: undefined"} |]
       { matchStatus = 406 }
+
+  context "media type parser allowed characters" $ do
+    it "regression test allowing charset=utf-8" $
+      request methodPost "/rpc/overloaded_default"
+        [("Content-Type", "application/json; charset=utf-8")]
+        [json|{"must_param":1}|]
+        `shouldRespondWith`
+        [json|{"val":1}|]
+        { matchStatus = 200
+        , matchHeaders = ["Content-Type" <:> "application/json; charset=utf-8"]
+        }
+
+    it "allows special chars as specified in RFC 2045 in the key and value of parameter" $
+      request methodPost "/rpc/overloaded_default"
+                                        --        key         =     value
+                                        --   ----------------- -----------------
+        [("Content-Type", "application/json; !#$%^&*-_+/'.{}~`=!#$%^&*-_+/'.{}~`")]
```

Comment:
If the key values are unknown, should we respond with an error? Or should we be lenient? Anything on the RFC about this?

Reply:
Not that anything should be changed now, but we could leave it as TODO or issue.. being too strict might break other clients.

Reply:
> Or should we be lenient? Anything on the RFC about this?

RFC says:
> MIME implementations must ignore any parameters whose names they do not recognize.

Thanks for pointing the leniency out. I actually wrote it to be lenient. The actual mistake was this:
```haskell
P.eof
```
After failing and backtracking, it looked for `EOF` and then failed. So, removing this one line should fix this issue entirely.